sdks/go/pkg/beam/core/graph/coder/row.go

```diff
@@ -24,13 +24,33 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+var (
+	defaultEnc RowEncoderBuilder
+	defaultDec RowDecoderBuilder
+
+	// RequireAllFieldsExported when set to true will have the default coder buildings using
+	// RowEncoderForStruct and RowDecoderForStruct fail if there are any unexported fields.
+	// When set false, unexported fields in default destination structs will be silently
+	// ignored when coding.
+	// This has no effect on types with registered coder providers.
+	RequireAllFieldsExported bool
```

Comment:
It seems like `RequireAllFieldsExported` would do better as an exported function that modifies the fields in `defaultEnc` and `defaultDec`. The current approach seems reliant on ordering in an unexpected way since this only gets set in the default enc/dec when `RegisterSchemaProviders` is called. So if you register some schema providers and then set this field to true afterwards, it won't have an effect.

Reply:
Good call. Or you set it and never register providers, it never gets set at all. Either way, not ideal.

Thanks!